test/test/optional.cpp

```diff
@@ -0,0 +1,22 @@
+#include "pch.h"
+#include "winrt/test_component.h"
+
+using namespace winrt;
+
+TEST_CASE("optional")
+{
+    test_component::Optional object;
+
+    object.Property(Windows::Foundation::IReference(123));
+    REQUIRE(object.Property().Value() == 123);
+
+    object.Property(nullptr);
+    REQUIRE(object.Property() == nullptr);
```

Comment:
Also

```cpp
REQUIRE(object.Property() == std::optional<int32_t>());
REQUIRE(object.Property() == std::nullopt);
```

to verify that implicit conversion to `std::optional<int32_t>` can successfully compare with std::nullopt.

Reply:
I don't think C++ supports this. The C++ compiler would have to perform an implicit `operator T` followed by an `operator ==` and that's more than C++ is willing to do implicitly. I would have to add `operator ==` for `std::optional` if this is desired.

Reply:
Okay well then nevermind then. May call out in a comment that those operations *don't* work.

Reply:
Let's consider adopting C++20's spaceship operator to simplify stuff like this.